### DISCUSSION TITLE
Find by guid?

### DISCUSSION THREAD

**Discussion by grantholle** [DirectoryTree/LdapRecord#498] [Q&A] [answered]
Is there a way that I can get the AD object from a guid? Here's what I've attempted on my `User` model:

```php
use LdapRecord\Models\ActiveDirectory\User as LdapUser;
use LdapRecord\Utilities;

LdapUser::findBy('objectguid', $this->guid); // I assumed this wouldn't work but tried anyway
LdapUser::findBy('objectguid', Utilities::stringGuidToHex($this->guid));
```

Both return `null`. I'd prefer to use the `guid` column on my `users` table rather than a `dn` or `mail` attribute...

**Comment by grantholle** [comment, marked as answer]
As is typical with me and posting to a discussion and then immediately finding it in the docs... There is a `findByGuid` function (as I would have guessed because @stevebauman is awesome).

**Reply by stevebauman** [reply]
Haha thanks @grantholle! I appreciate that 😄  I'm glad you were able to find it.

The search function of the docs site isn't as good as I'd like. It currently uses `Fuse.js` since Algolia Docsearch didn't support multiple versions at the time. I believe this is no longer the case, so I'm going to re-implement it soon so search is much more refined.
